src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverFactoryActivator.java

```diff
@@ -145,6 +145,11 @@ private static final class FactoryRegistration {
     /** Factory registration. */
     private volatile FactoryRegistration factoryRegistration;
 
+    private final VanityPathConfigurer vanityPathConfigurer = new VanityPathConfigurer();
```

Comment:
Why don't we make this a volatile field and `VanityPathConfigurer` (or preferably  `VanityPathConfiguration`) immutable?